tap_google_sheets/transform.py

```diff
@@ -145,37 +145,80 @@ def transform_sheet_boolean_data(value, unformatted_value, sheet_title, col_name
         return col_val
 
 # transform decimal values in the sheet
-def transform_sheet_decimal_data(value, sheet_title, col_name, col_letter, row_num, col_type):
+def transform_sheet_decimal_data(formatted_value, unformatted_value, sheet_title, col_name, col_letter, row_num, col_type):
+    """
+        Transform number type data and return according to the datatype in the sheet
+
+        :param
+        formatted_value - The displayed value of a cell in the sheet ie. 2022-01-01
+        unformatted_value - The calculated value of the field as per the value type ie. 44562,
+            (the date values are converted into serial numbers by Google's API)
+        sheet_title - The title of the sheet
+        col_name - Column name
+        col_letter - Column letter of the record ie. A, B, C, etc.
+        row_num - Row number of the record
+        col_type - Column type of the record (here: numberType)
+    """
+
+    # Removing comma to handle US number type format ie. 123,456.10 -> 123456.10
+    numeric_value = formatted_value.replace(",", "")
+    try:
+        # Verify we can convert formatted value to float for scientific formatted numbers
+        # For example:
+        #   formatted value: "1.23E+03"
+        #   unformatted value: 1234
+        # thus, we can convert "1.23E+03" to float but, for int casting we get error and wrong value will be returned
+        float(numeric_value)
+    except ValueError:
+        LOGGER.info('Received the value for sheet: {}, column: {}, cell: {}{} with unexpected data type. Ingesting this value with string format in the target'.format(
+            sheet_title, col_name, col_letter, row_num
+        ))
+        return str(formatted_value) # Return original value in case of ValueError
+
+    if type(unformatted_value) == int:
```

Comment:
Use `isinstance()` instead of `type()` for type checking in Python

Reply:
We used `type()` here because `isinstance()` will return true for boolean value in case of integer.

```
>>> isinstance(True, int)
True
```